textdb/textdb-dataflow/src/test/java/com/google/re2j/RE2JWrapperTest.java

```diff
@@ -0,0 +1,88 @@
+package com.google.re2j;
+
+import java.util.Arrays;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RE2JWrapperTest {
+	
+	private boolean subExprEqual(Regexp[] originalSubs, PublicRegexp[] compareSubs) {
+		if (originalSubs == null && compareSubs == null) {
+			return true;
+		} else if (originalSubs == null || compareSubs == null) {
+			return false;
+		} else {
+			if (originalSubs.length != compareSubs.length) {
+				return false;
+			} else {
+				boolean equivalent = true;
+				for (int i = 0; i < originalSubs.length; i++) {
+					equivalent = equivalent && isEquivalent(originalSubs[i], compareSubs[i]);
+				}
+				return equivalent;
+			}
+		}
+	}
+	
+	private boolean isEquivalent(Regexp original, PublicRegexp compare) {
+		boolean equivalent = 
+				(original.cap	== compare.cap) &&
+				(original.flags	== compare.flags) &&
+				(original.max	== compare.max) &&
+				(original.min	== compare.min) &&
+				(original.op	== compare.op) &&
+				(original.name == null ? compare.name == null : original.name.equals(compare.name)) &&
+				(Arrays.equals(original.runes, compare.runes)) &&
+				(subExprEqual(original.subs, compare.publicSubs));	
+		
+		return equivalent;
+	}
+	
+	private void testRE2Wrapper(String regex) {
```

Comment:
Add a comment to explain how we do testing: "We compare the results generated by our wrapper classes with those generated by the original classes."